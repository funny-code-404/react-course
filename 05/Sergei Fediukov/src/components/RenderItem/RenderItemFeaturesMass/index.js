const RenderItemFeaturesMass = (props) => {
    const { features_mass } = props
    return (
        features_mass.map(x => {
            for (const key in x) {
                return (
                    x[key] &&
                    <div key={key}>
                        <p >{key.charAt(0).toUpperCase() + key.substr(1).replace(/\_/g, ' ')} </p>
                        <ul>
                            {x[key].map((x) => <li key={x}>{x}</li>)}
                        </ul>
                    </div>
                )
            }
        })
    )
}
export default RenderItemFeaturesMass