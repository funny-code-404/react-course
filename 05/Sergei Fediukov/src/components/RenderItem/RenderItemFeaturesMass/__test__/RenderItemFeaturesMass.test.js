import { configure, shallow } from 'enzyme'
import Adapter from '@wojtekmaj/enzyme-adapter-react-17'

import RenderItemFeaturesMass from '..'

configure({ adapter: new Adapter() });

describe('Should render correctly', () => {
    const props = {
        features_mass: [{ test: ['test feature 1', 'test feature 2'] }],
    }
    let wrapper;
    beforeEach(() => {
        wrapper = shallow(<RenderItemFeaturesMass {...props} />)
    })
    it('to Match Snapshot', () => {
        expect(wrapper).toMatchSnapshot();
    })
    it('component is defined', () => {
        expect(wrapper).toHaveLength(1)
    })
})