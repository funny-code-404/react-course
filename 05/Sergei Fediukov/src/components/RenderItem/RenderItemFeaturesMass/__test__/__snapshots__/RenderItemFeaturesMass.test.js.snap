// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render correctly to Match Snapshot 1`] = `
<div
  key="test"
>
  <p>
    Test
     
  </p>
  <ul>
    <li
      key="test feature 1"
    >
      test feature 1
    </li>
    <li
      key="test feature 2"
    >
      test feature 2
    </li>
  </ul>
</div>
`;
